tests/unit/test_config.py

```diff
@@ -0,0 +1,352 @@
+OLSConfig:
+  {role}_provider: p1
+    """,
+            InvalidConfigurationError,
+            f"{role}_provider is specified, but {role}_model is missing",
+        )
+
+        check_expected_exception(
+            f"""
+---
+LLMProviders:
+  - name: p1
+    url: 'http://url1'
+    models:
+      - name: m1
+        url: 'http://murl1'
+OLSConfig:
+  {role}_model: m1
+    """,
+            InvalidConfigurationError,
+            f"{role}_model is specified, but {role}_provider is missing",
+        )
+
+    check_expected_exception(
+        """
+---
+LLMProviders:
+  - name: p1
+    url: 'http://url1'
+    models:
+      - name: m1
+        url: 'https://murl1'
+OLSConfig:
+  conversation_cache:
+    type: in-memory
+    redis:
+      url: 127.0.0.1:1234
+      credentials:
+          user: root
+          password: pwd123
+""",
+        InvalidConfigurationError,
+        "in-memory configuration is missing",
+    )
+
+    check_expected_exception(
+        """
+---
+LLMProviders:
+  - name: p1
+    url: 'http://url1'
+    models:
+      - name: m1
+        url: 'https://murl1'
+OLSConfig:
+  conversation_cache:
+    type: redis
+    in-memory:
+      max_entries: 1000
+""",
+        InvalidConfigurationError,
+        "redis configuration is missing",
+    )
+
+    check_expected_exception(
+        """
+---
+LLMProviders:
+  - name: p1
+    url: 'http://url1'
+    credentials: no_such_file_provider
+""",
+        FileNotFoundError,
+        "No such file or directory: 'no_such_file_provider'",
+    )
+
+    check_expected_exception(
+        """
+---
+LLMProviders:
+  - name: p1
+    url: 'http://url1'
+    credentials: secret.txt
+    models:
+      - name: m1
+        url: 'https://murl1'
+        credentials: no_such_file_model
+""",
+        FileNotFoundError,
+        "No such file or directory: 'no_such_file_model'",
+    )
+
+
+def test_valid_config_stream() -> None:
+    """Check if a valid configuration stream is handled correctly."""
+    try:
+        config.load_config_from_stream(
+            io.StringIO(
+                """
+---
+LLMProviders:
+  - name: p1
+    url: 'http://url1'
+    credentials: secret.txt
+    models:
+      - name: m1
+        url: 'http://murl1'
+        credentials: secret.txt
+      - name: m2
+        url: 'https://murl2'
+  - name: p2
+    url: 'https://url2'
+    models:
+      - name: m1
+        url: 'https://murl1'
+      - name: m2
+        url: 'https://murl2'
+OLSConfig:
+  enable_debug_ui: false
+  conversation_cache:
+    type: in-memory
+    in-memory:
+      max_entries: 1000
+  logger_config:
+    level: info
+  default_provider: p1
+  default_model: m1
+  classifier_provider: p2
+  classifier_model: m1
+"""
+            )
+        )
+    except Exception:
+        print(traceback.format_exc())
+        assert False
+
+
+def test_valid_config_file() -> None:
+    """Check if a valid configuration file is handled correctly."""
+    try:
+        config.init_config("olsconfig.yaml")
```

Comment:
if we're going to have a test based on a fixture data file, please create a directory structure for test data and put this there.

And then separately you can provide a sample config yaml in the top level of the repository